### PR TITLE
Reorganise LCLS-Tools modules for unit tests

### PR DESCRIPTION
**Pull request by MattKing06**
## Summary of Changes

The below reorganization may have an effect on codebases that utilize these modules. Please review your code after reading through these changes.

### Source Code Reorganization

As part of #84 it was decided that some cleaning up of the module structure was needed along with providing the unit-test directory structure.

In the top level directory, we still have:

```
lcls-tools
  -> docs
  -> examples
  -> lcls_tools
  ----> common
  ----> normalconducting
  ----> superconducting  
```

However, some submodules of the `common` module have now been reorganized:
```
-----> common
--------> beam_calcs (still the same)
--------> beam (not implemented yet)
-----------> transverse (not implemented yet)
-----------> longitudinal (not implemented yet) 
--------> controls
-----------> pyepics (formerly pyepics_tools)
--------> data_analysis (still contains archiver.py)
-----------> fitting (formerly math_tools)
--------> devices (still the same)
--------> frontend (formerly pydm_tools)
-----------> display (contains displayUtils, now known as util)
-----------> plotting (contains pydmPlotUtil, now known as util)
-----------> widgets (contains magnet_template.ui etc. to be use for more complex widget sharing)
--------> image_processing (still the same, needs to be dissected a bit more)
--------> logger (still the same)
--------> matlab2py (used for common dataset conversions from matlab applications to python types)
-----------> cor_plot (for correlation plot matlab files)
-----------> emit_scan (for emittance scan matlab files)
-----------> image (for images stored stored as matlab files)
-----> normalconducting (still the same)
-----> superconducting (still the same)
```

### Unit Test Directory Structure

We are using the `unittest` module in python standard library to conduct the testing of the lcls-tools module. To allow `unittest` to find the tests automatically, we must have a specified directory structure that mirrors the top-level source code directories.

Some of the previous unit tests were mixed in with the source code and they have been separated out into the test area and the associated datasets have been added to the `tests/datasets` sub-folders

The aim of this PR is not to make the tests pass, but to give us the ability to run all of the tests via `unittest`.

### To Test

- Go to the top level of this repository
- Run `python3 -m unittest discover'
- You should see that: 
  - 38 tests ran!
  - The majority of tests are skipped due to not finding the dataset files
  - 2 tests fail due to the expected values being tested not matching the dataset files

### CODE REVIEW

File: lcls_tools/common/devices/profile_monitor/profile_monitor.py [hunk 1]
```diff
@@ -146,7 +149,9 @@ def _collect_image_data(self, images, callback):
                 self._images.append(image)
                 sleep(delay)
                 i += 1
-        if callback:  # Would want this to be pyqtSignal or Event notification type thing
+        if (
+            callback
+        ):  # Would want this to be pyqtSignal or Event notification type thing
```
Comment (nneveu) [inline lcls_tools/common/devices/profile_monitor/profile_monitor.py:154]: Move this comment

File: lcls_tools/common/devices/profile_monitor/profile_monitor_test.py [hunk 2]
```diff
@@ -9,42 +9,38 @@
```
Comment (nneveu) [inline lcls_tools/common/devices/profile_monitor/profile_monitor_test.py:9]: move this to tests

File: lcls_tools/common/matlab2py/image/mat_image.py [hunk 3]
```diff
@@ -154,21 +160,24 @@ def _unpack_mat_data(self, mat_file):
...
 
     def load_mat_image(self, mat_file):
         """Converting super gross .mat image data structure to an object"""
+        if not os.path.isfile(mat_file):
+            raise FileNotFoundError(f"Could not find {mat_file}")
         try:
             self._unpack_mat_data(mat_file)
         except Exception as e:
-            print('error loading mat file: {0}'.format(e))
+            print("error loading mat file {0}: {1}".format(mat_file, e))
```
Comment (nneveu) [inline lcls_tools/common/matlab2py/image/mat_image.py:174]: Where did the {1} come from? I'm just not familiar with formatting strings for printing.

File: tests/datasets/images/matlab/junk.mat
Comment (nneveu) [inline tests/datasets/images/matlab/junk.mat:1]: Change names/info?

**Review by nneveu — CHANGES_REQUESTED** [review]

**Comment by nneveu** [comment]
Thanks all these changes, this is looking great!!!

File: lcls_tools/common/matlab2py/image/mat_image.py (hunk 3 above)
Comment (MattKing06) [reply lcls_tools/common/matlab2py/image/mat_image.py:174]: This looks like a python2 hangover to me. There are 2 arguments that get injected into the print statement and the 0,1 refer to their order in the .format function (0 = mat_file, 1 = exception-object)

File: lcls_tools/common/devices/profile_monitor/profile_monitor_test.py (hunk 2 above)
Comment (MattKing06) [reply lcls_tools/common/devices/profile_monitor/profile_monitor_test.py:12]: magnet_test.py is moved over as well

File: tests/datasets/images/matlab/junk.mat
Comment (MattKing06) [reply tests/datasets/images/matlab/junk.mat:1]: Contents now changed, hopefully a bit more informative but these files _really_ be corrupted/bad matlab files with actual data in

**Review by lisazacarias — APPROVED** [review]
My few files in the sea of changes look fine! Whenever this gets merged in I'll install a local copy and whack a mole my other repos

**Review by nneveu — APPROVED** [review]
Thanks for the reorg!

**Pull request merged**